fe/lib/components/LocationSuggest/LocationSuggest.tsx

```diff
@@ -60,7 +66,10 @@ export const LocationSuggest = forwardRef<HTMLInputElement, Props>(
         error: nearestLocationsError,
         loading: nearestLocationsLoading,
       },
-    ] = useLazyQuery(NEAREST_LOCATIONS, {
+    ] = useLazyQuery<
+      { nearestLocations: Maybe<Array<Location>> },
```

Comment:
AFAIK this isn't ideal because it isn't limited to the subset of fields you've selected in your query. This is what [TypeScript Operations](https://graphql-code-generator.com/docs/plugins/typescript-operations) will do, along with running your query against the SEEK API to ensure its validity

Reply:
Can you elaborate a bit more on this @72636c ?

Reply:
If you have a schema like this:

```graphql
type Query {
  location: Location!
}

type Location {
  id: String!
  description: String!
}
```

A type generated from the schema won't necessarily reflect what you've selectively retrieved in your query:

```typescript
// Type generated from the schema
interface Location {
  id: string
  description: string
}

// Notice that we do not select `description`
const QUERY = gql`
  query Location {
    location {
      id
    }
  }
`;

const { data } = useQuery<{ location: Location }>(QUERY);

// This is invalid but TypeScript doesn't catch it
data.description;
```


The TypeScript Operations plugin can inspect `QUERY` and produce a better type like this:

```typescript
// Type generated from `QUERY`
interface LocationQuery {
  location: Pick<Location, 'id'>;
}
```

Reply:
Gothcya. That makes sense. Do we use this anywhere in Wingman yet so I can see how it works?

Reply:
💥 Done! The types are dynamically typed using `typescript-operations`. We should do the same for `JobCategorySuggest` 

Tested on RA and everything looks ✅ 

@72636c

Reply:
P.S. I've just added the location queries as part of documents for now while generating types but we could do something similar to RA and nest it under `src/operations`